sktime/forecasting/base/adapters/_darts.py

```diff
@@ -0,0 +1,409 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements adapter for Darts models."""
+
+import abc
+from typing import Optional
+
+import pandas as pd
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+__author__ = ["yarnabrina", "fnhirwa"]
+
+
+class _DartsRegressionModelsAdapter(BaseForecaster):
+    """Base adapter class for Darts Regression models.
+
+    Parameters
+    ----------
+    past_covariates : Optional[List[str]], optional
+        column names in ``X`` which are known only for historical data, by default None
+    num_samples : Optional[int], optional
+        Number of times a prediction is sampled from a probabilistic model, by default
+        1000
```

Comment:
I assume you kept just these two following my old attempt at generic darts adapter at #5043. If you are focusing on regression models, similar to my another attempt at #5997, I think we can possibly consider a few more handling as part of base adapter itself. If all regression models in Darts inherits [RegressionModel](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.regression_model.html), this adapter may also do similar probably.

Reply:
yeah, I kept them from the initial implementation, this adapter implementation is of regression models but didn't cover the logic in https://github.com/sktime/sktime/pull/5997. I think a better option would be to keep the logic as it is and make a PR for [Regression Model](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.regression_model.html) that is extendable by these models. The downside is that there might be slight changes in this implementation if this adapter inherits from `RegressionModel` as done in darts.

Reply:
Sorry, may be there is a misunderstanding. I'm not suggesting you to inherit from darts class. I was suggesting to consider using all arguments of darts base class as arguments of your base adapter, if it makes sense. I did similarly in #5997, if you want an example. Let me know what you think.

> I think a better option would be to keep the logic as it is and make a PR for [Regression Model](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.regression_model.html) that is extendable by these models

Can you please elaborate? I didn't understand.

Reply:
> Sorry, may be there is a misunderstanding. I'm not suggesting you to inherit from darts class. I was suggesting to consider using all arguments of darts base class as arguments of your base adapter, if it makes sense. I did similarly in #5997, if you want an example. Let me know what you think.
> 
> > I think a better option would be to keep the logic as it is and make a PR for [Regression Model](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.regression_model.html) that is extendable by these models
> 
> Can you please elaborate? I didn't understand.

Okay now I get it 👍, I am going to try covering all the arguments